src/metadata/metadata_raw.c

```diff
@@ -380,7 +380,7 @@ static int _raw_ram_flush_do_asynch_fill(ocf_cache_t cache,
 	raw = ctx->raw;
 	ENV_BUG_ON(!raw);
 
-	size = raw->entry_size * raw->entries_in_page;
+	size = (uint64_t) raw->entry_size * raw->entries_in_page;
```

Comment:
`size` doesn't need to be 64-bit long. In fact total size of entries within page can never be bigger than 4096 bytes, so even `uint16_t` should be more than enough.